bpf/bpf_sock.c

```diff
@@ -389,6 +390,17 @@ static __always_inline int __sock4_xlate_fwd(struct bpf_sock_addr *ctx,
 				 */
 				backend_id = 0;
 		}
+	} else if (udp_only) {
+		udp_sock_affinity = true;
```

Comment:
Doesn't `udp_only` imply `udp_sock_affinity = true`?

Reply:
Also can we fold this whole path into the above? It's pretty much the same logic, except for the id population and `lb4_affinity_backend_id_by_netns` vs `lb4_affinity_backend_id_by_sock` retrieval. Maybe this could even be refactored as `lb4_affinity_backend_id(svc, &id, {SOCK,NETNS})`.

Reply:
Also, IPv6 bits are missing.

Reply:
Re IPv6 changes : I first wanted to check if I don't run into any verifier complexity issues on ipv4.